lib/matplotlib/cbook/__init__.py

```diff
@@ -2766,3 +2766,42 @@ def _topmost_artist(
     in reverse order.
     """
     return _cached_max(reversed(artists))
+
+
+def _define_aliases(local_d, alias_d):
+    """Define property aliases.
+
+    Use in a class definition as ::
+
+        cbook._define_aliases(locals(), {
+            "property": ["alias", ...], ...
+        })
+
+    For each property, if the corresponding ``get_property`` is defined in the
+    class so far, an alias named ``get_alias`` will be defined; the same will
+    be done for setters.  If neither the getter nor the setter exists, an
+    exception will be raised.
+
+    The alias map is stored as the ``_alias_map`` attribute on the class and
+    can be used by `~.normalize_kwargs`.
+    """
+
+    def make_alias(name):  # Enfore a closure over *name*.
+        def method(self, *args, **kwargs):
+            return getattr(self, name)(*args, **kwargs)
+        method.__doc__ = "alias for {}".format(name)
+        return method
```

Comment:
Why not do the `getattr` call outside of `method`?

Reply:
So that patching the base method gets propagated to the alias method.

Reply:
Ah, I see. That makes sense to me now